wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -537,13 +539,36 @@ public Revision commit(CommitParams params) throws GitException {
                 throw new GitException("Committer can't be null");
             }
 
-            //Check that there are staged changes present for commit, or any changes if is 'isAll' enabled, otherwise throw exception
             Status status = status(StatusFormat.SHORT);
-            if (!params.isAmend() && !params.isAll()
-                && status.getAdded().isEmpty() && status.getChanged().isEmpty() && status.getRemoved().isEmpty()) {
-                throw new GitException("No changes added to commit");
-            } else if (!params.isAmend() && params.isAll() && status.isClean()) {
-                throw new GitException("Nothing to commit, working directory clean");
+            List<String> staged = new ArrayList<>();
+            staged.addAll(status.getAdded());
+            staged.addAll(status.getChanged());
+            staged.addAll(status.getRemoved());
+            List<String> specified = params.getFiles();
+            List<String> changed = new ArrayList<>(staged);
+            changed.addAll(status.getModified());
+            changed.addAll(status.getMissing());
+            List<String> specifiedChanged = specified.stream()
+                                                     .filter(path -> changed.stream().anyMatch(c -> c.startsWith(path)))
+                                                     .collect(Collectors.toList());
+
+            // Check that specified paths don't contain untracked paths
+            String errorMessage = specified.stream()
+                                           .filter(path -> status.getUntracked().stream().anyMatch(u -> u.startsWith(path)))
+                                           .map(path -> format("error: pathspec '%s' did not match any file(s) known to git.\n", path))
+                                           .collect(joining());
+            if (!errorMessage.isEmpty()) {
+                throw new GitException(errorMessage);
+            }
+
+            // Check that there are changes present for commit, if 'isAmend' is disabled
+            if (!params.isAmend()) {
+                // Check that there are staged changes present for commit, or any changes if 'isAll' is enabled
+                if (status.isClean()) {
+                    throw new GitException("Nothing to commit, working directory clean");
```

Comment:
Should we throw exception that are related to changes and then about missing commiter's name and email?
Please check behavior of native git and do in the same way.
Any way I think we should check repository state and https://github.com/eclipse/che/pull/3833/files#diff-7f323c0167c366f41270f82a35ed368dR579 and only then throw exception related to changes.